master/pkg/searcher/tournament.go

```diff
@@ -139,11 +141,28 @@ func (s *tournamentSearch) Unit() expconf.Unit {
 }
 
 func (s *tournamentSearch) markCreates(subSearchID int, operations []Operation) []Operation {
+	res := []Operation{}
 	for _, operation := range operations {
 		switch operation := operation.(type) {
 		case Create:
 			s.TrialTable[operation.RequestID] = subSearchID
+			res = append(res, operation)
+		case Shutdown:
+			s.ShutdownSubSearches[subSearchID] = operation
+		default:
+			res = append(res, operation)
 		}
 	}
-	return operations
+	if len(s.ShutdownSubSearches) == len(s.subSearches) {
+		// does tournament fail iff any subsearch fails
+		// or does it fail iff all subsearches fail?
```

Comment:
The answer from Liam is `if any of the sub searchers failed`, so the current implementation is correct.

Reply:
in this case, @nrajanee, would you mind removing the comments and adding `break` after `failed = true`.